src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs

```diff
@@ -292,6 +311,146 @@ internal IKeyRing GetCurrentKeyRingCore(DateTime utcNow, bool forceRefresh = fal
         }
     }
 
+    private IKeyRing GetCurrentKeyRingCoreNew(DateTime utcNow, bool forceRefresh)
+    {
+        Debug.Assert(utcNow.Kind == DateTimeKind.Utc);
```

Comment:
Nit: A description or comment to clarify why this might not be the case?

Reply:
Sorry for not making it clearer in the review, but this method started as a copy-paste of the one above.  Unfortunately, since I left that method intact as an escape hatch, the diff doesn't reflect that.  And it's just a calling convention for this private member - it's documented as an assert, rather than a comment.

Reply:
Yeah, I observed that this was shared between the old and new implementations. I just didn't quite grok _why_ we needed to do this check at all.

Reply:
https://stackoverflow.com/questions/31220481/how-to-compare-datetime-values-taking-into-account-timezone

Reply:
I've added explanatory comments here and in the old method.